src/views/Instance/data.js

```diff
@@ -0,0 +1,146 @@
+    compoundsGroupFromDataset.forEach(function (compound) {
+      if (nodesParentsDict[compound.target] == null) nodesParentsDict[compound.target] = [];
+      nodesParentsDict[compound.target].push(compound.source);
+    });
+  });
+  return nodesParentsDict;
+};
+
+export const _processGraphNodes = (processedData, nodesParentsDict, dataset, nodesGroups) => {
+  Object.entries(nodesGroups).forEach(function ([nodesGroupName, nodesGroupMetadata]) {
+    // Nodes data
+    const nodesGroupFromDataset = dataset.content[nodesGroupName] || [];
+    nodesGroupFromDataset.forEach(function (node) {
+      processedData.graphElements.push(
+        _forgeCytoscapeNodeData(node, [nodesGroupName], nodesGroupMetadata, nodesParentsDict)
+      );
+    });
+    // Nodes style
+    processedData.stylesheet.push({
+      selector: `node.${nodesGroupName}`,
+      style: { ...DEFAULT_NODE_STYLE, ...nodesGroupMetadata.style },
+    });
+    processedData.stylesheet.push({
+      selector: `node.${nodesGroupName}:selected`,
+      style: { ...DEFAULT_SELECTED_NODE_STYLE, ...nodesGroupMetadata.style },
+    });
+  });
+};
+
+export const _processGraphEdges = (processedData, dataset, edgesGroups) => {
+  Object.entries(edgesGroups).forEach(function ([edgesGroupName, edgesGroupMetadata]) {
+    // Edges data
+    const edgesGroupFromDataset = dataset.content[edgesGroupName] || [];
+    edgesGroupFromDataset.forEach(function (edge) {
+      processedData.graphElements.push(_forgeCytoscapeEdgeData(edge, [edgesGroupName], edgesGroupMetadata));
+    });
+    // Edges style
+    processedData.stylesheet.push({
+      selector: `edge.${edgesGroupName}`,
+      style: { ...DEFAULT_EDGE_STYLE, ...edgesGroupMetadata.style },
+    });
+    processedData.stylesheet.push({
+      selector: `edge.${edgesGroupName}:selected`,
+      style: { ...DEFAULT_SELECTED_EDGE_STYLE, ...edgesGroupMetadata.style },
+    });
+  });
+};
+
+export const processGraphElements = (scenario) => {
+  if (!IS_INSTANCE_VIEW_FUNCTION_CONFIG_VALID) return [];
+  const processedData = {
+    graphElements: [],
+    stylesheet: [],
+  };
+  // Datasets can be found in scenario.datasets, scenario parameters can be found in scenario.parameters
+  for (const dataset of Object.values(scenario.datasets)) {
+    const nodesParentsDict = _processGraphCompounds(dataset, instanceViewData.dataContent.compounds || {});
+    _processGraphNodes(processedData, nodesParentsDict, dataset, instanceViewData.dataContent.nodes || {});
+    _processGraphEdges(processedData, dataset, instanceViewData.dataContent.edges || {});
+  }
+  return processedData;
+};
+
+async function _fetchDataFromADT(scenarioId, dataSource) {
+  const tokens = await Auth.acquireTokens();
+  const headers = { 'x-functions-key': dataSource.functionKey };
+  if (tokens?.accessToken) {
+    headers.common = {};
+    headers.common.Authorization = 'Bearer ' + tokens.accessToken;
+  }
+  return await axios({
```

Comment:
I've got an eslint warning about `return await`, maybe we can disable the rule if we need it ? [https://eslint.org/docs/rules/no-return-await](https://eslint.org/docs/rules/no-return-await)